tests/python/unittest/test_random.py

```diff
@@ -197,6 +197,75 @@ def test_random():
     check_with_device(mx.context.current_context(), 'float64')
 
 
+# Set seed variously based on `start_seed` and `num_init_seeds`, then set seed finally to `final_seed`
+def set_seed_variously(init_seed, num_init_seeds, final_seed):
+    end_seed = init_seed + num_init_seeds
+    for seed in range(init_seed, end_seed):
+        mx.random.seed(seed)
```

Comment:
What's the point of setting the seed multiple times? Do you expect a different behaviour depending on whether you set it once or multiple times?

Reply:
The problem being fixed by the PR is that the cpu pushes gpu seed-setting kernels to the engine, then can launch random-number-consuming kernels without synchronization (perhaps to a different stream).  Pushing multiple seed-setting kernels before the ultimate desired one is one way of aggravating this race condition.